### PR TITLE
[FEAT] Authentication Page ->  Verify code page

### LINKED ISSUE #23: [FEAT] Authentication Page -> Verify code page
### Description
Implement the verify code page as shown in the design

### Acceptance Criteria
- Responsive across all screen sizes (desktop, tablet and mobile)
- Write tests for your code.
- Use components where necessary (header, footer, buttons, modal and so on)

### Timeframe
24 hours

[Link to Design](https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=2-10859&t=p2gECcZQmYqvvrn7-4)

### Preview
image

### PR DESCRIPTION
**Pull request by Vixs101**
# Description

Implement the verify code page as shown in the design

figma link:
https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=2-10859&t=p2gECcZQmYqvvrn7-4

Closes #23 

# Changes proposed

## What were you told to do?
* make corrections on the previous pr

[FEAT] Authentication Page -> Verify code page

## What did you do?

* Created a the route for the verify-otp page as suggested in the ROUTING.md file
* implemented the verify-otp page 

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/Videos

Screenshot from 2024-07-21 16-50-26
Screenshot from 2024-07-21 18-45-02

### CODE REVIEW

File: src/app/settings/layout.tsx [hunk 1]
```diff
@@ -15,6 +15,6 @@ export default function SettingsLayout({
           {children}
         </div>
       </section>
-    </>
+    </StateContextProvider>
```
[the remark this answers is not in the document]
Comment (Vixs101) [reply src/app/settings/layout.tsx]: Please i didn't work on this file, it was gotten from the dev branch that way

File: src/test/verify-code.test.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    expect.assertions(2);
+    render(<VerifyCodePage />);
+    expect(screen.getByText("ellafedora@gmail.com")).toBeInTheDocument();
+    expect(screen.getByText("00:59")).toBeInTheDocument();
+  });
+
+  it("renders the OTP input component", () => {
+    expect.assertions(0);
+    render(<VerifyCodePage />);
+    const otpInputs = screen.getAllByRole("textbox");
```
[the remark this answers is not in the document]
Comment (Vixs101) [reply src/test/verify-code.test.tsx]: Alright, i'm working on them now

**Comment by Vixs101** [comment]
Please the test and lint failure are from a file already merged to the dev branch not mine

**Comment by ObiFaith** [comment]
I am not a mentor, Victor. I can’t review your PR

Try to fix the eslint error else you might not get your PR merge

**Comment by Vixs101** [comment]
> I am not a mentor, Victor. I can’t review your PR
> 
> Try to fix the eslint error else you might not get your PR merge

Thanks alot, I'll do that right away

**Pull request closed without merging**